src/esportsbot/cogs/MusicCog.py

```diff
@@ -0,0 +1,1117 @@
+        """
+        threads = min(20, len(links))
+        with concurrent.futures.ThreadPoolExecutor(max_workers=threads) as executor:
+            generator = executor.map(self.__get_basic_song_information, links)
+            return generator
+
+    async def __get_basic_song_information(self, url):
+        """
+        Download the title for a given url.
+        :param url: The url to find the title of.
+        :return: A dict with link and title keys.
+        """
+
+        # Using a random user agent means there is less chance of being blocked by google
+        user_agents = ['Mozilla/5.0 CK={} (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko',
+                       'Mozilla/5.0 (Windows NT 5.1; rv:7.0.1) Gecko/20100101 Firefox/7.0.1',
+                       'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) '
+                       + 'Chrome/74.0.3729.169 Safari/537.36',
+                       'Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1; SV1; .NET CLR 1.1.4322)',
+                       'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:40.0) Gecko/20100101 Firefox/40.1',
+                       'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) '
+                       + 'Chrome/74.0.3729.157 Safari/537.36']
+        shuffle(user_agents)
+        # Wait some random time if its been less than 2 seconds since last ping
+        if self._youtube_last_pinged - time.time() < 2:
+            await asyncio.sleep(float(2) + uniform(0.1, 2.1))
+        headers = {'User-Agent': user_agents[0], 'Referer': 'https://www.google.co.uk/'}
+
+        resp = requests.get(url, headers=headers)
+        text = resp.text
+        page_title = text[text.find('<title>') + 7:text.find('</title>')]
+        formatted_title = html.unescape(page_title).replace(' - YouTube', '')
+        self._youtube_last_pinged = time.time()
+        return formatted_title
+
+    def __check_loops_alive(self):
+        """
+        Check if the async task loops are alive and start any that are not.
+        """
+        if not self.check_active_channels.is_running():
+            self.check_active_channels.start()
+        if not self.check_marked_channels.is_running():
+            self.check_marked_channels.start()
+        if not self.reset_music_allowance.is_running():
+            self.reset_music_allowance.start()
+
+    def __add_new_active_channel(self, guild_id, channel_id=None, voice_client=None) -> bool:
+        """
+        Add a new voice channel to the currently active channels.
+        :param guild_id: The id of the guild the voice channel is in.
+        :param channel_id: The id of the voice channel the bot is joining.
+        :param voice_client: The voice client instance of the bot.
+        :return: True if successfully added to the list of active channels. False otherwise.
+        """
+
+        if guild_id not in self._currently_active:
```

Comment:
Instead of:
```
        if guild_id not in self._currently_active:
            # If the guild is not currently active we can add it
            self._currently_active[guild_id] = {}
            self._currently_active[guild_id]['channel_id'] = channel_id
            self._currently_active[guild_id]['voice_client'] = voice_client
            self._currently_active[guild_id]['queue'] = []
            self._currently_active[guild_id]['current_song'] = None
```

I would do:
```
        if guild_id not in self._currently_active:
            # If the guild is not currently active we can add it
            self._currently_active[guild_id] = {'channel_id': channel_id, 'voice_client': voice_client,
                                                                 'queue': [], 'current_song': None}
```